src/Raven.Server/Documents/ETL/Providers/AI/Embeddings/EmbeddingsGenerationScriptTransformer.cs

```diff
@@ -0,0 +1,493 @@
+{
+    private readonly EmbeddingsGenerationConfiguration _configuration;
+    private EmbeddingsGenerationScriptRun _currentRun;
+    private readonly PatchRequest _mainScript;
+
+    public EmbeddingsGenerationScriptTransformer(DocumentDatabase database, DocumentsOperationContext context, Transformation transformation, PatchRequest behaviorFunctions, EmbeddingsGenerationConfiguration configuration) : base(database, context, null, behaviorFunctions)
+    {
+        _configuration = configuration;
+        _mainScript = new PatchRequest(transformation.Script, PatchRequestType.EmbeddingsGeneration);
+    }
+
+    public override void Initialize(bool debugMode)
+    {
+        Database.Scripts.GetScriptRunner(_mainScript, true, out DocumentScript);
+
+        if (DocumentScript == null)
+            return;
+
+        if (debugMode)
+            DocumentScript.DebugMode = true;
+
+        ObjectInstance embeddingsObject = new JsObject(DocumentScript.ScriptEngine);
+        embeddingsObject.FastSetProperty("generate", new PropertyDescriptor(new ClrFunction(DocumentScript.ScriptEngine, "generate", EmbeddingsGenerate), false, false, false));
+        DocumentScript.ScriptEngine.SetValue("embeddings", embeddingsObject);
+
+        ObjectInstance textObject = new JsObject(DocumentScript.ScriptEngine);
+        textObject.FastSetProperty("split", new PropertyDescriptor(new ClrFunction(DocumentScript.ScriptEngine, "split", SplitPlainText), false, false, false));
+        textObject.FastSetProperty("splitLines", new PropertyDescriptor(new ClrFunction(DocumentScript.ScriptEngine, "splitLines", SplitPlainTextLines), false, false, false));
+        textObject.FastSetProperty("splitParagraphs", new PropertyDescriptor(new ClrFunction(DocumentScript.ScriptEngine, "splitParagraphs", SplitPlainTextParagraphs), false, false, false));
+        DocumentScript.ScriptEngine.SetValue("text", textObject);
+
+        ObjectInstance markdownObject = new JsObject(DocumentScript.ScriptEngine);
+        markdownObject.FastSetProperty("splitLines", new PropertyDescriptor(new ClrFunction(DocumentScript.ScriptEngine, "splitLines", SplitMarkDownLines), false, false, false));
+        markdownObject.FastSetProperty("splitParagraphs", new PropertyDescriptor(new ClrFunction(DocumentScript.ScriptEngine, "splitParagraphs", SplitMarkDownParagraphs), false, false, false));
+        DocumentScript.ScriptEngine.SetValue("markdown", markdownObject);
+
+        ObjectInstance htmlObject = new JsObject(DocumentScript.ScriptEngine);
+        htmlObject.FastSetProperty("strip", new PropertyDescriptor(new ClrFunction(DocumentScript.ScriptEngine, "strip", StripHtml), false, false, false));
+        DocumentScript.ScriptEngine.SetValue("html", htmlObject);
+    }
+
+    protected override void AddLoadedAttachment(JsValue reference, string name, Attachment attachment)
```

Comment:
I think we should be able to load an attachment in the ETL. That is likely to be very useful

Reply:
Yes, we should try to provide that ability. At glance it shouldn't be an issue. Opened a ticket for that: https://issues.hibernatingrhinos.com/issue/RavenDB-23865/AI-Integration-Embeddings-Generation-Ability-to-load-vector-from-attachment-in-JS-script